src/lib/utils.py

```diff
@@ -109,26 +110,40 @@ async def get_extra_data_info_from_url(
     dest_io: t.Optional[t.IO] = None,
     content_type_deny: t.Optional[t.Iterable[re.Pattern]] = None,
 ):
+
     async with session.get(
         url,
         skip_auto_headers=[aiohttp.hdrs.ACCEPT_ENCODING],
     ) as response:
         real_url = str(response.url)
         info = response.headers
 
-        content_type = info.get(aiohttp.hdrs.CONTENT_TYPE)
+        http_content_type = info.get(aiohttp.hdrs.CONTENT_TYPE)
         if (
-            content_type is not None
+            http_content_type is not None
             and content_type_deny is not None
-            and any(r.match(content_type) for r in content_type_deny)
+            and any(r.match(http_content_type) for r in content_type_deny)
         ):
-            raise CheckerFetchError(
-                f"Wrong content type '{content_type}' received from '{url}'"
+            log.debug(
+                f"Wrong http content type '{http_content_type}' received from '{url}'"
             )
 
+        first_chunk = True
         checksum = MultiHash()
         size = 0
         async for chunk in response.content.iter_chunked(HTTP_CHUNK_SIZE):
+            if first_chunk:
```

Comment:
I think we should try to detect content type from response body only when we previously got unexpected one in http header.

Reply:
You mean if this condition in line 122 is false?

            http_content_type is not None
            and content_type_deny is not None
            and any(r.match(http_content_type) for r in content_type_deny)

Reply:
Yes, so that the logic becomes something like as follows: ff content type from headers matched denylist, then try to determine content type from the first chunk of data (and if it matched again, raise an error).
I guess you can put the denylist match check to a nested function like
```python
async def get_extra_data_info_from_url(...):
    def content_type_rejected(content_type: t.Optional[str]) -> bool:
        return (
            content_type is not None
            and content_type_deny is not None
            and any(r.match(content_type) for r in content_type_deny)
        )
    ...
```